qcodes/instrument_drivers/Lakeshore/Model_325.py

```diff
@@ -1,11 +1,87 @@
 import numpy as np
-from typing import cast
+from typing import cast, Union, List, Tuple, Iterable
+from itertools import takewhile, repeat
+import os
 
 from qcodes import VisaInstrument, InstrumentChannel, ChannelList
 from qcodes.utils.validators import Enum, Numbers
 from qcodes.instrument.group_parameter import GroupParameter, Group
 
 
+def read_curve_file(file_path: str) -> dict:
+    """
+    Read a curve file with extension *.330
+    The format of the file is as follows:
+
+    Item1: value
+    Item2: value
+    Item3: value
+
+    Header1   Header2   Header3
+    data11     data12    data13
+    data21     data22    data23
+    """
+
+    def split_data_line(
+            line: str,
+            parsers: Union[type, List]=str
+    ) -> List[str]:
+
+        if not isinstance(parsers, list):
+            parsers = repeat(parsers)
+
+        line_split = [i for i in line.split("  ") if i != ""]
+        return [
+            parser(i) for parser, i in zip(parsers, line_split)
+        ]
+
+    def strip(strings: Iterable[str]) -> Tuple:
+        return tuple(s.strip() for s in strings)
+
+    with open(file_path, "r") as curve_file:
+        lines = iter(curve_file.readlines())
+
+    # Meta data lines contain a colon
+    metadata_lines = takewhile(lambda s: ":" in s, lines)
+    # Data from the file is collected in the following dict
+    file_data = dict([strip(line.split(":")) for line in metadata_lines])
+    # After meta data we have a data header
+    header_items = strip(split_data_line(next(lines)))
+    # After that we have the curve data
+    data = [
+        split_data_line(line, parsers=float)
+        for line in lines if line.strip() != ""
+    ]
+
+    file_data["data"] = dict(
+        zip(header_items, zip(*data))
+    )
+
+    return file_data
+
+
+def sanitize_data(file_data: dict) -> None:
+    """
+    Data as found in the curve files are slightly different then
+    the dictionary as expected by the 'upload_curve' method of the
+    driver
+
+    Note: This function modifies the input
+    """
+    data_dict = file_data["data"]
+    # We do not need the index column
+    del data_dict["No."]
+    # Rename the 'Units' column to the appropriate name
+    # Look up under the 'Data Format' entry to find what units we have
+    data_format = file_data['Data Format']
+    # This is a string in the form '4      (Log Ohms/Kelvin)'
+    data_format_int = int(data_format[0])
```

Comment:
what is the total number of supported units? is it indeed 4 as `valid_sensor_units` suggests? regardless, i'd make this line a bit more robust via `... = int(data_format.split()[0])`, is it fair to do so?